paasta_tools/mesos_maintenance.py

```diff
@@ -377,7 +377,11 @@ def hostnames_to_components(hostnames, resolve=False):
             (host, ip) = hostname.split('|')
             components.append(Hostname(host=host, ip=ip))
         else:
-            ip = gethostbyname(hostname) if resolve else None
+            try:
+                ip = gethostbyname(hostname) if resolve else None
+            except Exception:
+                ip = None
+                log.error(f"Failed to resolve IP for {hostname}, continuing regardless")
```

Comment:
Wait wait, are you sure you want to append still though?

Reply:
You're right, no need to actually append. Let me fix this up!